Dockerfile.base

```diff
@@ -28,36 +28,40 @@ RUN apt-get update && \
     ninja-build \
     cmake
 
-# gclient dependencies
+# python dependencies
 RUN pip3 install httplib2 six
 
+# set gcc-8 as default
 RUN update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-8 0
 RUN update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-8 0
 
-# clone depot_tools and set path
+# clone depot_tools and add it to your path
 RUN git clone https://chromium.googlesource.com/chromium/tools/depot_tools.git
 
-# build gn and run tests
+# build gn and run tests (running tests is optional)
+# you may need to checkout older versions of gn if it's being used to build older versions of V8
+# use a platform specific `ninja` binary
 RUN git clone https://gn.googlesource.com/gn && \
     cd gn && \
-    python build/gen.py && \
+    CXX=g++ python build/gen.py && \
     ninja -C out && \
     out/gn_unittests && \
     cp /home/gn/out/gn /bin/gn && \
     chmod +x /bin/gn
 
-# set env vars and args
+# set environment variables
 ENV PATH=$PATH:/home/depot_tools/
 ENV VPYTHON_BYPASS="manually managed python not supported by chrome operations"
 ENV V8_BRANCH="main"
 ENV MODE="release"
 
+# copy bin folder and any patches needed by jenkins
+# (this step may be optional when bulding V8 manually)
```

Comment:
We should be consistent with how things are tagged as optional.  Maybe worthwhile just prefixing the steps with `(Optional): Copy bin folder and patches needed by Jenkins CI/CD.  Not needed when building V8 manually`

And update the next two changes to prefix with `(Optional):` too.

Reply:
Thanks Joran, I have updated the comments. 
Richard pointed out that having a Red Hat licence header on an Ubuntu based Dockerfile may look odd, we can add it if you think it can be helpful.

Reply:
The copyright protects the contents of the file, not the base image used by the Dockerfile itself. :)

Reply:
I think it'll ultimately be what the Red Hat legal process dictates.  For reference, we have the license/copyright in our Java Ubuntu based docker files: https://github.com/ibmruntimes/ci.docker/blob/master/ibmjava/8/sdk/ubuntu/Dockerfile

Reply:
Sounds good, added headers to both our dockerfiles.